include/CL/sycl/detail/small_array.hpp

```diff
@@ -272,6 +290,26 @@ struct small_array : std::array<BasicType, Dims>,
 
 };
 
+#define TRISYCL_OPERATOR_SIZE_T_OP(op)                                \
+template <typename BasicType, typename FinalType, std::size_t Dims>   \
+struct small_array<BasicType, FinalType, Dims> operator op(           \
+  const size_t &lhs,                                                  \
+  const struct small_array<BasicType, FinalType, Dims> &rhs)          \
+{                                                                     \
+  FinalType res;                                                      \
+  for (std::size_t i = 0; i != Dims; ++i)                             \
+    res[i] = lhs op rhs[i];                                           \
+  return res;                                                         \
+}
+
+/* Comparison operators without boost as boost only specs bool returns, and the sycl spec doesn't want those */
+TRISYCL_OPERATOR_SIZE_T_OP(>)
+TRISYCL_OPERATOR_SIZE_T_OP(<)
+TRISYCL_OPERATOR_SIZE_T_OP(<=)
+TRISYCL_OPERATOR_SIZE_T_OP(>=)
```

Comment:
No `==`?

Reply:
== and != are covered by the std::array comparable and the spec never calls them out as needing overloads operators, so I assume the work with bool returns.